app/controllers/product_users_controller.rb

```diff
@@ -69,14 +69,15 @@ def destroy
 
   # PUT /facilities/:facility_id/instruments/:instrument_id/update_restrictions
   def update_restrictions
-    product_param_name = @product.class.name.underscore
-    unless params[product_param_name]
+    permitted_params = params.require(@product.class.name.underscore).require(:product_users)
+
+    unless permitted_params
       redirect_to action: :index
       return
     end
-    params[product_param_name][:product_users].each do |key, value|
+    permitted_params.each do |key, value|
```

Comment:
I remember there being some confusion around this. Would it be worth renaming `key` and `value` to whatever they actual are? `product_user_id`, I think?